### ISSUE TITLE
[syncfusion_flutter_charts] series set to [isVisible: false] shows up after setState

### ISSUE THREAD

**Issue by FisherWL** [syncfusion/flutter-widgets#521]
``` dart
import 'package:flutter/material.dart';
import 'package:syncfusion_flutter_charts/charts.dart';

void main() => runApp(MaterialApp(home: MyApp()));

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: Column(
        children: [
          ElevatedButton(
              child: Text('setState'),
              onPressed: () {
                setState(() {});
              }),
          SfCartesianChart(
            primaryXAxis: CategoryAxis(title: AxisTitle(text: 'Primary X Axis')),
            series: <ChartSeries>[
              LineSeries<SalesData, String>(
                dataSource: [
                  SalesData('Jan', 35),
                  SalesData('Feb', 28),
                  SalesData('Mar', 34),
                  SalesData('Apr', 32),
                  SalesData('May', 40),
                ],
                xValueMapper: (SalesData sales, _) => sales.year,
                yValueMapper: (SalesData sales, _) => sales.sales,
                isVisible: false,
              )
            ],
          ),
        ],
      ),
    );
  }
}

class SalesData {
  SalesData(this.year, this.sales, [this.numeric]);
  final String year;
  double? sales;
  final double? numeric;
}

```

![2022-01-04 01 38 02](https://user-images.githubusercontent.com/57275868/148039474-5ba31edf-c871-4539-874c-5194c46590b2.gif)

**Comment by FisherWL** [comment]
my testing shows the issue is related to

` legend: Legend(isVisible: false)`
- if default or set to false, the issue recurs

also if series set to `isVisibleInLegend: false`
the issue recurs too.

**Comment by Yuvaraj-Gajaraj** [comment]
Hi @FisherWL,

We have analyzed your query and we can replicate the reported scenario at our end. This will be fixed and included in our upcoming weekly patch release on 18th January 2022, and we appreciate your patience until then.

Regards,
Yuvaraj.

**Comment by FisherWL** [comment]
@Yuvaraj-Gajaraj
the setstate issue is fixed with 19.4.43, thx,
but 19.4.43 disabled the override method drawDataMarker as the links below mentioned:
[https://github.com/syncfusion/flutter-widgets/issues/494](url)
[https://github.com/syncfusion/flutter-widgets/issues/539](url)

**Comment by Yuvaraj-Gajaraj** [comment]
Hi FisherWL,

We have already fixed the drawDataMarker override method issue and it will be rolled out in our Vol 4 SP1 release, so we will update once it gets rolled out in the following issue https://github.com/syncfusion/flutter-widgets/issues/539. So, you can follow this henceforth.

Regards,
Yuvaraj.
